source/extensions/filters/common/ext_authz/check_request_utils.h

```diff
@@ -70,8 +70,10 @@ class CheckRequestUtils {
   static void setAttrContextPeer(envoy::service::auth::v3alpha::AttributeContext::Peer& peer,
                                  const Network::Connection& connection, const std::string& service,
                                  const bool local, bool include_certificate);
+  static void setRequestTime(envoy::service::auth::v3alpha::AttributeContext::Request& req,
+                             Envoy::Http::StreamDecoderFilterCallbacks* callbacks);
   static void setHttpRequest(envoy::service::auth::v3alpha::AttributeContext::HttpRequest& httpreq,
-                             const Envoy::Http::StreamDecoderFilterCallbacks* callbacks,
+                             Envoy::Http::StreamDecoderFilterCallbacks* callbacks,
```

Comment:
Why do the callbacks become mutable?

Reply:
Before this change, inside that setHttpRequest we had `const_cast` https://github.com/envoyproxy/envoy/pull/9678/files#diff-a1cbd6e906473b8ff3d53eac314d8213L99-L100 in order to access `streamId()` and `streamInfo()` since "'this' argument to member function 'streamId' has type 'const Envoy::Http::StreamDecoderFilterCallbacks', but function is not marked const". In this change, I need to access `streamInfo()` once again to get the `startTime`.

Reply:
Is there a reason we can't provide `const` ID/info methods? Would prefer that if possible.

Reply:
Can do with `streamId()`. The `streamInfo()` is mutable, e.g. when route uses it to `setResponseFlag`.